crates/wasmtime/src/func.rs

```diff
@@ -440,7 +444,15 @@ impl Func {
             "cannot use `new_async` without enabling async support in the config"
         );
         Func::new(store, ty, move |mut caller, params, results| {
-            let async_cx = caller.store.as_context_mut().0.async_cx();
+            let async_cx =
+                caller
+                    .store
+                    .as_context_mut()
+                    .0
+                    .async_cx()
+                    .ok_or(anyhow::format_err!(
+                        "Attempt to spawn new action on dying fiber"
+                    ))?;
```

Comment:
Same as above I think this is best done as `.expect()` since it would be bad if wasm were able to execute and only caught it when an import happened.

Reply:
Addressed in a775e97.